src/lazy/text/buffer.rs

```diff
@@ -383,38 +404,42 @@ impl<'data> TextBufferView<'data> {
             // If the next thing in the input is a `}`, return `None`.
             value(None, Self::match_struct_end),
             // Otherwise, match a name/value pair and turn it into a `LazyRawTextField`.
-            Self::match_struct_field_name_and_value.map(
```

Comment:
🗺️ As you'll see below, `clippy` (rightfully) complained about the complexity of the type returned by `match_struct_field_name_and_value` among others. I've introduced a `MatchedFieldName` type that simplifies the signatures quite a bit.